python/tvm/autotvm/task/dispatcher.py

```diff
@@ -481,8 +412,12 @@ def _query_inside(self, target, workload):
         """
         if self._counter < len(self._records):
             cfg = self._records[self._counter][0].config
+            wkl = self._records[self._counter][0].task.workload
+            if workload is not None:
+                assert wkl == workload
             self._counter += 1
-            self.update(target, workload, cfg)
+            self.update(target, wkl, cfg)
+            cfg.workload = wkl
```

Comment:
Where is `cfg.workload` initialized? I didn't find a definition in `ConfigSpace`. Also what's the purpose to have a workload field in a config space?

Reply:
This is only specific to `ApplyGraphBest`. The reason is complication. Because `ApplyGraphBest` relies on the order of query, we cannot use `relay.backend.compile_engine.select_implement` to collect the autotvm workload as it may query more than once. Therefore, this is a temporary work around that we sneak in the workload in the return cfg. We can remove this part of logic after we make `ApplyGraphBest` no longer relies on the query order.
@kevinthesun

Reply:
I see. Could we define `self.workload` in `ConfigSpace` and add your comment on it? So that we will remember to remove it in the future.